src/main/java/com/mraof/minestuck/entity/animation/PhasedMobAnimation.java

```diff
@@ -13,47 +20,61 @@ public class PhasedMobAnimation
 	private final int contactStart;
 	private final int recoveryStart;
 	private final int recoveryEnd;
+	private final double baseAnimationSpeed;
+	@Nullable
+	private final Attribute speedModifyingAttribute;
+	
+	public PhasedMobAnimation(MobAnimation animation, int initiationStart, int contactStart, int recoveryStart, double baseAnimationSpeed)
+	{
+		this(animation, initiationStart, contactStart, recoveryStart, baseAnimationSpeed, Attributes.ATTACK_SPEED);
+	}
+	
 	
 	/**
 	 * @param initiationStart not the first frame of animation
 	 * @param contactStart the apex of animations, when attacks connect
+	 * @param baseAnimationSpeed how fast the animation is played normally
+	 * @param speedModifyingAttribute the entity attribute responsible for affecting the animation's speed
 	 */
-	public PhasedMobAnimation(MobAnimation animation, int initiationStart, int contactStart, int recoveryStart)
+	public PhasedMobAnimation(MobAnimation animation, int initiationStart, int contactStart, int recoveryStart, double baseAnimationSpeed, @Nullable Attribute speedModifyingAttribute)
 	{
 		this.animation = animation;
-		this.initiationStart = initiationStart;
-		this.contactStart = contactStart;
-		this.recoveryStart = recoveryStart;
-		this.recoveryEnd = animation.animationLength(); //recoveryEnd is identical to animationLength in MobAnimation
+		this.initiationStart = (int) Math.round(initiationStart * baseAnimationSpeed);
+		this.contactStart =  (int) Math.round(contactStart * baseAnimationSpeed);
+		this.recoveryStart =  (int) Math.round(recoveryStart * baseAnimationSpeed);
+		this.recoveryEnd =  (int) Math.round(animation.animationLength() * baseAnimationSpeed); //recoveryEnd is identical to animationLength in MobAnimation
+		this.baseAnimationSpeed = baseAnimationSpeed;
+		this.speedModifyingAttribute = speedModifyingAttribute;
 	}
 	
-	public Phases getCurrentPhase(int time)
+	public Phases getCurrentPhase(PathfinderMob entity, int time)
 	{
-		if(time < initiationStart)
+		double speed = getAnimationSpeed(entity);
+		if(time < getInitiationStartTime(speed))
 			return Phases.ANTICIPATION;
-		else if(time < contactStart)
+		else if(time < getContactStartTime(speed))
 			return Phases.INITIATION;
-		else if(time < recoveryStart)
+		else if(time < getRecoveryStartTime(speed))
 			return Phases.CONTACT;
-		else if(time < recoveryEnd)
+		else if(time < getTotalAnimationLength(speed))
 			return Phases.RECOVERY;
 		else
 			return Phases.NEUTRAL;
 	}
 	
-	public int getInitiationStartTime()
+	public int getInitiationStartTime(double speed)
 	{
-		return initiationStart;
+		return (int) Math.round(initiationStart / speed);
 	}
 	
-	public int getContactStartTime()
+	public int getContactStartTime(double speed)
 	{
-		return contactStart;
+		return (int) Math.round(contactStart / speed);
 	}
 	
-	public int getRecoveryStartTime()
+	public int getRecoveryStartTime(double speed)
 	{
-		return recoveryStart;
+		return (int) Math.round(recoveryStart / speed);
 	}
```

Comment:
I see calls like `phasedAnimation.getContactStartTime(phasedAnimation.getAnimationSpeed(entity))` in a lot of places. It would be nice with an entity overload for these so that you only need to call on `phasedAnimation` once:
`phasedAnimation.getContactStartTime(entity)`
Depending on where the double overloads would still be used, they could potentially be privated or even replaced.